RestaurantReservation.Db/Services/CustomerService.cs

```diff
@@ -14,34 +14,34 @@ public CustomerService(RestaurantReservationDbContext context)
             _customersRepository = new(_context);
```

Comment:
Services should never have access to the db context at all, the idea of the repos interfaces is to use them to talk to DB without the need to directly know the implementation of DB layer nor the need to expose the db context to the other layers. 

Let's please remove this context from all services, and instead of `private CustomerRepository _customersRepository;` do `private ICustomerRepository _customersRepository;`. And you need to inject all repos with their interfaces into IoC as you did with the services. This way the repo constructor will automatically have the dbcontext without the need to pass down to it.